src/vgr_cairo.mli

```diff
@@ -0,0 +1,117 @@
+(*---------------------------------------------------------------------------
+   Copyright 2014 Arthur Wendling, Daniel C. Bünzli. All rights reserved.
+   Distributed under the BSD3 license, see license at the end of the file.
+   %%NAME%% release %%VERSION%%
+  ---------------------------------------------------------------------------*)
+
+(* Based on the Vgr_htmlc documentation by Daniel C. Bünzli. *)
+
+(** Vg Cairo renderer.
+
+    {b Dependencies:}
+    {ul {- {e {{:http://forge.ocamlcore.org/projects/cairo/}Cairo2 bindings
+     for OCaml}}}
+     {- {e {{:http://cairographics.org/}Cairo Graphics library}}}}
+
+    {e Release %%VERSION%% — %%MAINTAINER%% } *)
+
+(** {1:target Cairo render targets} *)
+
+val target : ?resolution:Gg.V2.t -> [< `PDF | `PNG | `PS | `SVG ] ->
+  Vg.Vgr.dst_stored Vg.Vgr.target
+(** [target resolution fmt] is a render target for rendering to the stored
+    destination given to {!Vg.Vgr.create} in the chosen format [fmt].
+    {ul
+    {- [resolution], specifies the rendering resolution in samples per
+       meters. The PDF, PS and SVG formats are measured in points by Cairo,
+       while the PNG format is in pixels. If unspecified, the default
+       conversion to points is used.}}
+
+    {b Multiple images.} Multiple images render on the target are not
+    supported. [Invalid_argument] is raised by {!Vg.Vgr.render} if multiple
+    images are rendered. *)
+
+val target_surface : ?size:Gg.size2 -> Cairo.Surface.t ->
+  [`Other] Vg.Vgr.target
+(** [target_surface size s] is a render target for rendering to the Cairo
+    surface [s].
+    {ul
+    {- The physical size of {{!Vg.Vgr.renderable}renderables} is ignored and
+       the view rectangle is mapped on the surface size.}
+    {- [size], Surfaces created with [Cairo.Surface] have a valid size, while
+       file based surfaces have a size of zero by default: If the size of the
+       surface can not be determined, the optional argument [size] is used
+       instead. [Invalid_argument] is raised if the size is invalid.}}
```

Comment:
> The PDF and PS surfaces do support resizing (but I think it would be confusing to arbitrarily resize for them.)

Shouldn't we rather resize according to the renderable's size ? That's what for example vg's built-in PDF renderer does.

> The four remaining modules allow you to create a file-based surface. For unknown reasons, this surfaces report a size of (0,0) when asked (so the ?size argument is used.)

If this is a bug I'm somewhat reluctant to adapt the API for the bug...